src/commands/CommandRouter.ts

```diff
@@ -0,0 +1,150 @@
+import { Client, Events, GuildScheduledEvent, Interaction, Message, REST } from "discord.js";
+import { RESTPostAPIApplicationCommandsJSONBody, Routes } from "discord-api-types/v9";
+import { Command, CommandDictionary } from "./Command";
+import ReactionCommand from "./reaction/ReactionCommand";
+import { Environment, UserIDs } from "../utils/constants";
+import { isModReaction } from "../utils/helpers";
+import CreateServerEvent from "./reaction/CreateServerEvent";
+import DiscordEventRouter from "../discord/DiscordEventRouter";
+import ContentCommand from "./messageContent/ContentCommand";
+import DiscordBot from "../discord/DiscordBot";
+
+interface CommandRouterConfiguration {
+    slashCommands: Command[],
+    reactionCommands: ReactionCommand[],
+    contentCommands: ContentCommand[]
+};
+
+export default class CommandRouter {
+    private _discordBot: DiscordBot;
+    private _eventRouter: DiscordEventRouter;
+
+    constructor(eventRouter: DiscordEventRouter, discordBot: DiscordBot, configuration: CommandRouterConfiguration) {
+        this._eventRouter = eventRouter;
+        this._discordBot = discordBot;
+        this.initialize(configuration);
+    }
+
+    private async initialize(configuration: CommandRouterConfiguration) {
+        this.registerSlashCommands(configuration.slashCommands);
+        this.listenForReactionCommands(configuration.reactionCommands);
+        this.listenForContentCommands(configuration.contentCommands);
```

Comment:
From an organizational perspective, it might be helpful to separate out the registerSlashCommands, listenForReactionCommands, and listenForContentCommands into their own classes and just instantiate/use them in here as the controller. That would make this class a bit "cleaner".

Reply:
Moved them out to their own files, definitely good for SRP.